Heffner_Daniel/Ruby Assignments/bank_account.rb

```diff
@@ -0,0 +1,62 @@
+class BankAccount
+	attr_reader :acct_num, :checking, :savings
+	
+	@@num_of_accts = 0
+	
+	def initialize
+		@acct_num = generate_acct_num
+		@checking = 0
+		@savings = 0
+		@@num_of_accts += 1
+	end
+	
+	def deposit amount, account
+		return "Please use a valid account type!" unless account == "checking" or account == "savings"
+		@checking += amount if account == "checking"
+		@savings += amount if account == "savings"
+	end
+	
+	def withdraw amount, account
+		return "Please use a valid account type!" unless account == "checking" or account == "savings"
+		case account
+		when "checking"
+			overdraft = @checking - amount
```

Comment:
I like where you are going, I'd suggest not nesting case statements. I'd change this and keep just the main case statement, see what you think:

return "Error" if @checking < amount 
return "#{@checking -=amount} withdrawn"